sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/models/KeyVaultRoleScope.java

```diff
@@ -30,10 +30,13 @@ public static KeyVaultRoleScope fromString(String name) {
      *
      * @param url A string representing a URL containing the name of the scope to look for.
      * @return The corresponding {@link KeyVaultRoleScope}.
-     * @throws MalformedURLException If the given {@link String URL String} is malformed.
      */
-    public static KeyVaultRoleScope fromUrl(String url) throws MalformedURLException {
-        return fromString(new URL(url).getPath(), KeyVaultRoleScope.class);
+    public static KeyVaultRoleScope fromUrl(String url) {
+        try {
+            return fromString(new URL(url).getPath(), KeyVaultRoleScope.class);
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
```

Comment:
Should still document that an exception is thrown by the method even if it isn't checked. Also, should this be an `IllegalArgumentException` to be more targeted?

Reply:
I think `IllegalArgumentException` is fine but since it a static method I'm not sure if( we should use a logger or not. I don't think we want to have static loggers. Any thoughts @srngar?

Reply:
Not using a logger for now, but did change to the exception type you suggested

Reply:
IllegalArgumentException should be logged and thrown here.